src/cargo/util/network/retry.rs

```diff
@@ -317,3 +358,57 @@ fn curle_http2_stream_is_spurious() {
     let err = curl::Error::new(code);
     assert!(maybe_spurious(&err.into()));
 }
+
+#[test]
+fn retry_after_parsing() {
+    use crate::core::Shell;
+    fn spurious(code: u32, header: &str) -> HttpNotSuccessful {
+        HttpNotSuccessful {
+            code,
+            url: "Uri".to_string(),
+            ip: None,
+            body: Vec::new(),
+            headers: vec![header.to_string()],
+        }
+    }
+
+    let headers = spurious(429, "Retry-After: 10");
+    assert_eq!(Retry::parse_retry_after(&headers), Some(10_000));
+
+    let headers = spurious(429, "retry-after: Fri, 01 Jan 2100 00:00:00 GMT");
+    let expected = jiff::Zoned::now()
+        .until(
+            &jiff::civil::date(2100, 1, 1)
+                .at(0, 0, 0, 0)
+                .to_zoned(jiff::tz::TimeZone::UTC)
+                .unwrap(),
+        )
+        .unwrap()
+        .total(jiff::Unit::Millisecond)
+        .unwrap() as u64;
+    let actual = Retry::parse_retry_after(&headers).unwrap();
+    let diff = expected.abs_diff(actual.into());
+    assert!((diff < 1000), "{} != {} ({})", expected, actual, diff);
```

Comment:
Independent of the problem with jiff initialization, I would recommend moving the `expected` initialization to *after* the `actual` line. That should reduce the time variance (since computing `expected` is very efficient, whereas `parse_retry_after` is not).

Reply:
I'll change `parse_retry_after` to take in `now` as a parameter to simplify the testing and eliminate the variation.

Reply:
Correction, I think I had that backwards. It looks like `jiff::Zoned::now()` is what is taking all the time.

Reply:
Can you try again with latest push and see if it's fixed?

Reply:
I reported this upstream at https://github.com/BurntSushi/jiff/issues/366.